Wire Notification Service Extension/NotificationService.swift

```
@@ -17,30 +17,65 @@
 //
 
 import UserNotifications
+import WireNotificationEngine
+import WireCommonComponents
+import WireDataModel
+import WireRequestStrategy
+import WireSyncEngine
 
-class NotificationService: UNNotificationServiceExtension {
-
+public class NotificationService: UNNotificationServiceExtension {
     var contentHandler: ((UNNotificationContent) -> Void)?
     var bestAttemptContent: UNMutableNotificationContent?
+    var notificationSession: NotificationSession?
 
-    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+    public override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        if notificationSession == nil {
+            notificationSession = try? self.createNotificationSession()
+        }
+        
         self.contentHandler = contentHandler
         bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
-        
-        if let bestAttemptContent = bestAttemptContent {
-            // Modify the notification content here...
-            bestAttemptContent.title = "\(bestAttemptContent.title) [modified]"
-            
-            contentHandler(bestAttemptContent)
-        }
     }
     
-    override func serviceExtensionTimeWillExpire() {
+    public override func serviceExtensionTimeWillExpire() {
```

If we don't manage to download the notification we still not show the original notification since it won't make sense to the user. We could either show some fallback message  *or* we hide the notification.

This is something we should discuss with product/design. I'd say add a `TODO` about this and hide the notification for now.

---

Done

---

We are still not hiding the notification by setting an empty body and title.

---

Ah yes...
I moved the logic from the part where we receive empty events, so there won't be any notifications.